### DISCUSSION TITLE
Mass assign @input.debounce to 100+ input elements to a page

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3575] [1. Help] [answered]
I want to mass assign `@input.debounce="foo($event)"` to a 100 input elements on a page.

```
inputs[i].onclick = function() {
    // Do something when the element is clicked
  };
```

But I need AlpineJS's data and I need debounce - `@input.debounce="saveLocally()"`

**Comment by ekwoka** [comment]
okay, so....do this:

```html
<input @input.debounce="saveLocally()" />
```

??

**Reply by anjanesh** [reply]
I was thinking a one-liner in AlpineJS to do this :

```
const sectionForm = document.getElementById('sectionForm');
const inputElements = sectionForm.querySelectorAll('input');
inputElements.forEach((input) =>
{
    input.setAttribute('@input.debounce.1000', 'saveLocally()');
});
```

**Reply by ekwoka** [reply, marked as answer]
you'll need to use `x-on:input` just because `@` is not a valid attribute technically, though this may be patched in 13.12.1

But why can't you just put it on the markup itself?

**Reply by anjanesh** [reply]
_But why can't you just put it on the markup itself?_ - There are ~100 input elements. The form is quite complex.

**Reply by ekwoka** [reply]
I...don't see why that matters?

But also, `input` events bubble, so you can just wrap the entire form in one listener

**Reply by anjanesh** [reply]
This is what I got from ChatGPT :
```
// Use Alpine.js to dynamically assign the event handler
document.addEventListener('alpine:init', () =>
{
    Alpine.directive('input-debounce', (el, { expression }, { evaluateLater }) =>
    {
        const saveLocally = evaluateLater(expression);
        el.addEventListener('input', () =>
        {
            setTimeout(() =>
            {
                realSaveFoo();
            }, 1000);
        });
    });

    // Alpine.start();
});

document.addEventListener("DOMContentLoaded", () =>
{
    const sectionForm = document.getElementById('section-form');
    const inputElements = sectionForm.querySelectorAll('input');
    inputElements.forEach((input) =>
    {
        input.setAttribute('x-input-debounce', 'saveLocally()');
    });
});

let realSaveFoo = () =>
{
    // Do the job of saving
});
```

But if replacing `input.setAttribute('@input.debounce.1000', 'saveLocally()');` with `x-on:input` works then it would be better, yet to try that.

**Reply by ekwoka** [reply]
No, you can just wrap the whole form in a single listener.

The event bubbles up, so you don't need it on each element.
